### DISCUSSION TITLE
Can vector ignore configuration errors on launch?

### DISCUSSION THREAD

**Discussion by Quyzi** [vectordotdev/vector#13484] [Q&A] [answered]
Can I configure vector to ignore errors in some configuration files on launch? 

I launch vector with `/usr/bin/vector --config-dir /etc/vector --watch-config` as a kube sidecar.

Use case here is I have one main configuration that provides common sinks and transforms that changes infrequently at `/etc/vector/main.yaml`. I also have many `/etc/vector/*.yaml` configuration files that do change frequently, and rely on the main configuration.  

A bad configuration file would cause the container to fail and the pods would crash loop. I'd like to launch vector to only require `main.yaml` to be valid, and ignore config files with errors. Can this be done?

**Comment by neuronull** [comment, marked as answer]
Hi @Quyzi !

Unfortunately currently vector requires the entire configuration to be valid.

I'd recommend opening a feature request for the ability to ignore configuration errors on parts of the config.
